test/e2e/e2e_test.go

```diff
@@ -591,6 +624,70 @@ func pollForResource(t *testing.T, resourceID string, timeout time.Duration, acc
 	return obj, err
 }
 
+func pollForUpdatedServerCertFromClient(t *testing.T, client *kubernetes.Clientset, config *rest.Config, namespace, serviceName, updatedServerCert string) error {
```

Comment:
Rather than creating a pod and reading data from it, how about the following?

- use alpine or busybox
- construct a command for the pod that directly tests for the expected cert
  - e.g. `bash -c "test <expected cert> == $(command to retrieve cert)"`
- check the exit code of the pod

Simpler and a good opportunity to exercise your bash-foo. :)

Reply:
> use alpine or busybox

maybe i did not search enough and this could done with alpine or busybox images. from my understanding, `openssl` is a good way to get server certs. I am not sure how to do this with other tools. Any pointers are very welcome 

> construct a command for the pod that directly tests for the expected cert

I am looking into this

Reply:
I pushed some changes. it does not entirely depend on bash script but will use a blend of both of bash commands and pod logs. please review